app/assets/src/components/views/SampleViewV2/ReportTable.jsx

```diff
@@ -340,6 +354,137 @@ class ReportTable extends React.Component {
     );
   };
 
+  // Hover actions
+  // TODO(tiago): consider moving the action inside HoverActions
+  linkToNCBI = ({ taxId }) => {
+    let num = parseInt(taxId);
+    if (num < -1e8) {
+      num = -num % -1e8;
+    }
+    num = num.toString();
+    const ncbiLink = `https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=${num}`;
+    window.open(ncbiLink, "hide_referrer");
+  };
+
+  downloadFastaUrl = ({ taxLevel, taxId }) => {
+    const { pipelineVersion, sampleId } = this.props;
+    location.href = `/samples/${sampleId}/fasta/${taxLevel}/${taxId}/NT_or_NR?pipeline_version=${pipelineVersion}`;
+  };
+
+  downloadContigUrl = ({ taxId }) => {
+    const { pipelineVersion, sampleId } = this.props;
+    location.href = `/samples/${sampleId}/taxid_contigs?taxid=${taxId}&pipeline_version=${pipelineVersion}`;
+  };
+
+  handleCoverageVizClick = ({
+    taxId,
+    taxLevel,
+    taxName,
+    taxCommonName,
+    taxSpecies,
+  }) => {
+    const { onCoverageVizClick, pipelineVersion, sampleId } = this.props;
+    const alignmentVizUrl = `/samples/${sampleId}/alignment_viz/nt_${taxLevel}_${taxId}?pipeline_version=${pipelineVersion}`;
+
+    if (pipelineVersionHasCoverageViz(pipelineVersion)) {
+      onCoverageVizClick({
+        taxId,
+        taxName,
+        taxCommonName,
+        taxLevel,
+        alignmentVizUrl,
+        taxSpecies,
+      });
+    } else {
+      window.open(alignmentVizUrl);
+    }
+  };
+
+  handlePhyloTreeModalOpen = phyloTreeModalParams => {
+    this.setState({
+      phyloTreeModalParams,
+    });
+  };
+
+  handlePhyloTreeModalClose = () => {
+    this.setState({
+      phyloTreeModalParams: null,
+    });
+  };
+
+  renderHoverActions = ({ rowData }) => {
+    const {
+      alignVizAvailable,
+      fastaDownloadEnabled,
+      phyloTreeAllowed,
+      pipelineVersion,
+      projectId,
+      sampleId,
+    } = this.props;
+
+    const validTaxId =
+      rowData.taxId < INVALID_CALL_BASE_TAXID || rowData.taxId > 0;
+    const contigVizEnabled = !!keys(getOr({}, "nt.contigs", rowData)).length;
+    const coverageVizEnabled =
+      alignVizAvailable && validTaxId && getOr(0, "nt.count", rowData) > 0;
+    const phyloTreeEnabled =
+      phyloTreeAllowed &&
+      rowData.taxId > 0 &&
+      PhyloTreeChecks.passesCreateCondition(
+        getOr(0, "nt.count", rowData),
+        getOr(0, "nr.count", rowData)
+      );
+
+    const analyticsContext = {
+      projectId: projectId,
```

Comment:
Previous version of the report includes `projectName` as well, do we want to keep that?

Reply:
No, it was conscious decision.